lib/oauth2/access_token.ex

```diff
@@ -1,6 +1,42 @@
 defmodule OAuth2.AccessToken do
   @moduledoc """
-  Provides functionality to make authorized requests to an OAuth2 provider.
+  This module defines the `OAuth2.AccessToken` struct and provides functionality
+  to make authorized requests to an OAuth2 provider using the AccessToken
+  returned by the provider.
+
+  The `OAuth2.AccessToken` struct is created for you when you use the
+  `OAuth2.Client.get_token`
+
+  ### Notes
+
+  * If a full url is given (e.g. "http://www.example.com/api/resource") then it
+  will use that otherwise you can specify an endpoint (e.g. "/api/resource") and
+  it will append it to the `Client.site`.
+
+  * The headers from the `Client.headers` are appended to the request headers.
+
+  ### Examples
+
+  ```
+  token = "abc123"
```

Comment:
Let's not use pipes here and below. Just show them fully qualified.

```elixir
token = OAuth2.AccessToken.new("abc123", client)
```

Also, for the future, (stylistically) when piping...move the first value to the next line and indent:

```elixir
something =
  something_else
  |> some_cool_function()
  |> a_really_awesome_function()
```